dask/blockwise.py

```diff
@@ -244,6 +265,62 @@ def _out_numblocks(self):
         return out_d
 
 
+class BlockwiseGetitem(Blockwise):
+    """
+    Blockwise dedicated to `__getitem__`
+
+    Parameters
+    ----------
+    getitem_key : Scalar or BlockwiseFilter
+    """
+
+    def __init__(
+        self,
+        output,
+        output_indices,
+        dsk,
+        indices,
+        numblocks,
+        concatenate=None,
+        new_axes=None,
+        getitem_key=None,
```

Comment:
After looking through the code for a bit I'm still not sure I understand exactly how this keyword is supposed to work.

Reply:
I'm not really happy with the name here. The intent is to capture the value `'B'` in the expression

```python
df['B']
```

But it's not restricted to being a simple scalar. It can also be a more complicated expression:

```python
In [24]: df = dd.read_parquet("/tmp/data.parquet/", engine="pyarrow")

In [25]: df['B'].dask.layers[df.B._name].getitem_key
Out[25]: 'B'

In [26]: df2 = df[df.B == 1]

In [27]: df2.dask.layers[df2._name].getitem_key
Out[27]: BlockwiseFilter<(('getitem-3a656fd14824e63ee9448be31f58035b', ('.0',)), (1, None)) -> eq-b37a6a08cbf5864e07ef16e236c2e22c>

In [28]: df2.dask.layers[df2._name].getitem_key.filter_key
Out[28]: (<function _operator.eq(a, b, /)>, 1)
```

Happy to have better names here.